Add stokes flow example

sifanw094 opened the pull request:
<!-- markdownlint-disable MD013-->
# Modulus Pull Request

## Description
<!-- Provide a standalone description of changes in this PR. -->
<!-- Reference any issues closed by this PR with "closes #1234". -->
<!-- Note: The pull request title will be included in the CHANGELOG. -->

## Checklist

- [x] I am familiar with the [Contributing Guidelines](https://github.com/NVIDIA/modulus/blob/main/CONTRIBUTING.md).
- [ ] New or existing tests cover these changes.
- [x] The documentation is up to date with these changes.
- [ ] The [CHANGELOG.md](https://github.com/NVIDIA/modulus/blob/main/CHANGELOG.md) is up to date with these changes.
- [x] An [issue](https://github.com/NVIDIA/modulus/issues) is linked to this pull request.

## Dependencies

<!-- Call out any new dependencies needed if any -->

sifanw094 commented:
blossom-ci

sifanw094 commented:
blossom-ci

mnabian commented:
/blossom-ci

mnabian commented:
@sifanw094 please add unit tests for this. You can take a look at the Ahmed body dataset tests as a sample.
Also please update the changelog.

ktangsali commented on modulus/datapipes/gnn/stokes_dataset.py:
```diff
@@ -0,0 +1,306 @@
+
+
+class StokesDataset(DGLDataset):
+    """In-memory Stokes flow Dataset"""
```
Can you add more docs here? Especially information on the parameters would be nice. Please refer: https://github.com/NVIDIA/modulus/blob/main/modulus/datapipes/gnn/ahmed_body_dataset.py#L62

sifanw094 replied:
Thank you for this comments! I have added detailed descritpion for this dataset.

ktangsali commented on modulus/datapipes/gnn/stokes_dataset.py:
```diff
@@ -0,0 +1,306 @@
+    @staticmethod
+    def denormalize(invar, mu, std):
+        """denormalizes a tensor"""
+        # assert invar.size()[-1] == mu.size()[-1]
```
Are these comments needed?

sifanw094 replied:
Thanks for pointing this out. No, this is not necessary and I have removed it.

sifanw094 commented:
> @sifanw094 please add unit tests for this. You can take a look at the Ahmed body dataset tests as a sample.
> Also please update the changelog.

Thanks for this comment. I have updated the changelog and added the test for Stokes flow dataset.

ktangsali commented on CHANGELOG.md:
```diff
@@ -10,6 +10,8 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ### Added
 
+- Stokes flow dataset
```
Very minor. To be consistent with other changelog additions, can you please rewrite this to "Added Stokes flow dataset"?

ktangsali reviewed (approved):
Thanks for the changes @sifanw094 left a minor comment about the changelog. Once that is done, it should be ready to merge.

ktangsali commented:
/blossom-ci

ktangsali commented:
/blossom-ci

ktangsali commented:
/blossom-ci
